### PR TITLE
main page UI

### PR DESCRIPTION
**Pull request by Omid07**

### CODE REVIEW

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
-    android:layout_height="match_parent"
-    android:paddingBottom="@dimen/activity_vertical_margin"
-    android:paddingLeft="@dimen/activity_horizontal_margin"
-    android:paddingRight="@dimen/activity_horizontal_margin"
-    android:paddingTop="@dimen/activity_vertical_margin"
-    tools:context="com.example.omid.rss_5.MainActivity">
+    android:layout_height="match_parent">
+
+    <android.support.v7.widget.RecyclerView
+        android:id="@+id/my_recycler_view"
```
Comment (ahammad-reyad) [inline app/src/main/res/layout/activity_main.xml]: use meaningful id

File: app/src/main/java/com/example/omid/rss_5/CategoryAdapter.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        holder.textShortDescription.setText(R.string.description);
+        holder.textDate.setText(R.string.date);
+    }
+
+    @Override
+    public int getItemCount() {
+        return mDataset.size();
+    }
+
+    public class ViewHolder extends RecyclerView.ViewHolder {
```
Comment (ahammad-reyad) [inline app/src/main/java/com/example/omid/rss_5/CategoryAdapter.java]: move ViewHolder to different file

File: app/src/main/java/com/example/omid/rss_5/CategoryAdapter.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        return viewHolder;
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        CategoryViewHolder viewHolder = (CategoryViewHolder) holder;
+        configNewsHolder(viewHolder, position);
+    }
+
+    private void configNewsHolder(CategoryViewHolder viewHolder, int position){
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/rss_5/CategoryAdapter.java]: why not build a method name such as update( item_in_your_list) and make set text value in ViewHolder???

File: app/src/main/java/com/example/omid/rss_5/CategoryViewHolder.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+    public CategoryViewHolder(View view) {
+        super(view);
+        textCategory = (TextView) view.findViewById(R.id.text_category);
+        textTitle = (TextView) view.findViewById(R.id.text_title);
+        textShortDescription = (TextView) view.findViewById(R.id.text_short_description);
+        textDate = (TextView) view.findViewById(R.id.text_date);
+    }
+
+    public TextView getCategory() {
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/rss_5/CategoryViewHolder.java]: 24,28, 32, 36 should be replace by a method update only to set their value

File: app/src/main/java/com/example/omid/rss_5/MainActivity.java [hunk 5]
```diff
@@ -1,13 +1,33 @@
...
+        mRecyclerView = (RecyclerView) findViewById(R.id.category_recycler_view);
+        Resources resources = getResources();
+        String [] categories = resources.getStringArray(R.array.categories);
+        final ArrayList<String> list = new ArrayList<>();
+        for (int i = 0; i < categories.length; ++i) {
+            list.add(categories[i]);
+        }
+        mLayoutManager = new LinearLayoutManager(this);
+        mRecyclerView.setLayoutManager(mLayoutManager);
+        mAdapter = new CategoryAdapter(list);
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/rss_5/MainActivity.java]: you can pass String [] to adapter not need to change to ArrayList<String> (2tat)

**Comment by nvmanh** [comment]
OK!

**Pull request merged**